lib/facility_factory.rb

```diff
@@ -0,0 +1,52 @@
+require "spec_helper"
+
+class FacilityFactory
```

Comment:
This class looks great! It's essentially what I was referring to in your `VehicleFactory` class. Break out the work a little bit. It's easier to read and test. I'd just pull the `create_facilities` method up top since it's the primary method, and then put the helpers below.